src/osd/ECBackend.h

```diff
@@ -351,13 +360,15 @@ class ECBackend : public PGBackend {
     std::map<pg_shard_t, std::vector<std::pair<int, int>>> need;
     bool want_attrs;
     GenContext<std::pair<RecoveryMessages *, read_result_t& > &> *cb;
+    bool partial_read;
     read_request_t(
       const std::list<boost::tuple<uint64_t, uint64_t, uint32_t> > &to_read,
       const std::map<pg_shard_t, std::vector<std::pair<int, int>>> &need,
       bool want_attrs,
-      GenContext<std::pair<RecoveryMessages *, read_result_t& > &> *cb)
+      GenContext<std::pair<RecoveryMessages *, read_result_t& > &> *cb,
+      bool partial_read=false)
```

Comment:
Why is this a parameter?  Wouldn't we always want to do a partial read if possible?  Why would we ever read more chunks than are necessary?

Reply:
Oops, isn't this a code code? I don't see any caller of `read_request_t()` setting it.